libraries/app/database_api.cpp

```diff
@@ -2061,48 +2061,43 @@ vector<blinded_balance_object> database_api_impl::get_blinded_balances( const fl
 //                                                                  //
 //////////////////////////////////////////////////////////////////////
 
-vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, uint32_t start, uint32_t limit)const
+vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
 {
    return my->get_withdrawals_giver( account, start, limit );
 }
 
-vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, uint32_t start, uint32_t limit)const
+vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
 {
    FC_ASSERT( limit <= 100 );
    vector<withdraw_permission_object> result;
 
-   auto withdraw_range = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().equal_range(account);
-   auto total = distance(withdraw_range.first, withdraw_range.second);
-   if(start >= total ) return result;
-   advance(withdraw_range.first, start);
-
-   std::for_each(withdraw_range.first, withdraw_range.second,
-                 [&result] (const withdraw_permission_object& withdraw) {
-                    result.emplace_back(withdraw);
-                 });
-
+   auto withdraw_index_end = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().end();
+   auto withdraw_itr = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().lower_bound(boost::make_tuple(account, start));
+   while( withdraw_itr->withdraw_from_account == account && result.size() < limit && withdraw_itr != withdraw_index_end)
```

Comment:
`end` check should be before dereferencing (`withdraw_itr->`)

Reply:
sorry, changed.